pkg/cloudprovider/gce/gce.go

```diff
@@ -177,7 +178,11 @@ func newGCECloud(config io.Reader) (*GCECloud, error) {
 			projectID = cfg.Global.ProjectID
 		}
 		if cfg.Global.NetworkName != "" {
-			networkName = cfg.Global.NetworkName
+			if strings.Contains(cfg.Global.NetworkName, "/") {
```

Comment:
Is looking for a slash enough to be convinced that you should use the global config value? A stronger check would be to look for the https://www.googleapis.com/ prefix (since this is in the gce package that should be safe).